src/core/extensions/extensionsmanager.cpp

```diff
@@ -0,0 +1,190 @@
+/*
+ * This file is part of Liri Browser
+ *
+ * Copyright (C) 2017 Tim Süberkrüb <tim.sueberkrueb@web.de>
+ *
+ * $BEGIN_LICENSE:GPL3+$
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * $END_LICENSE$
+*/
+
+#include "extensionsmanager.h"
+
+#include <QDebug>
+
+
+ExtensionsManager::ExtensionsManager(QObject *parent)
+    : QObject(parent)
+{
+    m_dir = QDir(Paths::ExtensionsLocation);
+
+    // Check for paths
+    if (!QDir(m_dir).exists()) {
+        qDebug() << "Extensions location doesn't exist.";
+        qDebug() << "Creating" << Paths::ExtensionsLocation << "...";
+        QDir().mkpath(Paths::ExtensionsLocation);
+    }
+
+    // Create models
+    m_extensionsModel = new ExtensionsModel(this);
+    extensionsModelChanged(m_extensionsModel);
+    m_extensionThemesModel = new ExtensionThemesModel(this);
+    extensionThemesModelChanged(m_extensionThemesModel);
+    m_extensionSearchEnginesModel = new ExtensionSearchEnginesModel(this);
+    extensionSearchEnginesModelChanged(m_extensionSearchEnginesModel);
+
+    // Create watcher
+    m_watcher = new QFileSystemWatcher(this);
+}
+
+void ExtensionsManager::loadBuiltins()
+{
+    Extension* extension = new Extension(this);
```

Comment:
If I got it right, for each extension there should be Extension parser. But extension parser can be reused to parse multiple extensions. Now it's not the case.